core/consensus/src/adapter.rs

```diff
@@ -387,6 +407,189 @@ where
         self.mempool
             .set_args(timeout_gap, cycles_limit, max_tx_size);
     }
+
+    /// make sure that the mempool is in the same height!
+    async fn verify_txs(&self, ctx: Context, txs: Vec<Hash>) -> ProtocolResult<()> {
+        self.mempool.ensure_order_txs(ctx.clone(), txs).await
+    }
+
+    /// this function verify all info in header except proof and roots
+    async fn verify_block_header(&self, ctx: Context, block: &Block) -> ProtocolResult<()> {
+        let previous_block = self
+            .get_block_by_height(ctx.clone(), block.header.height - 1)
+            .await?;
+
+        let previous_block_hash = Hash::digest(previous_block.encode_fixed()?);
+
+        if previous_block_hash != block.header.pre_hash {
+            return Err(ConsensusError::VerifyBlockHeaderPreBlockHash(block.header.height).into());
+        }
+
+        // the block 0 and 1 's proof is consensus-ed by community
+        if block.header.height > 1u64 && block.header.pre_hash != block.header.proof.block_hash {
```

Comment:
block 1 has proof.

Reply:
block_1's proof has no sig and bitmap, cann't verify